### PR TITLE
Hts 761

### PR DESCRIPTION
**Pull request by Jackie1988**

### CODE REVIEW

File: app/uk/gov/hmrc/helptosaveapi/ApplicationRegistration.scala [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.helptosaveapi
+
+import java.util.concurrent.TimeUnit
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/helptosaveapi/ApplicationRegistration.scala]: unused import

File: app/uk/gov/hmrc/helptosaveapi/ApplicationRegistration.scala [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+                                         config:                  Configuration)(implicit ec: ExecutionContext) extends Logging {
+
+  val registrationEnabled: Boolean = config.underlying.getBoolean("service-locator-registration.enabled")
+
+  val duration: FiniteDuration = config.underlying.get[FiniteDuration]("service-locator-registration.delay").value
+
+  val cancellable: Cancellable = application.actorSystem.scheduler.scheduleOnce(duration,
+    new Runnable {
+      override def run(): Unit = if (registrationEnabled) serviceLocatorConnector.register().onComplete{
+        case Success(result) ⇒
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/helptosaveapi/ApplicationRegistration.scala]: i guess `result` is boolean, so it registers the service successfully only if its `true`  - 
what about if its `false ? `

File: app/uk/gov/hmrc/helptosaveapi/connectors/ServiceLocatorConnector.scala [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+package uk.gov.hmrc.helptosaveapi.connectors
+
+import javax.inject.Inject
+
+import cats.instances.int._
+import cats.syntax.eq._
+import com.google.inject.{ImplementedBy, Singleton}
+import play.api.Mode.Mode
+import play.api.{Configuration, Environment}
+import play.api.http.{ContentTypes, HeaderNames}
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/helptosaveapi/connectors/ServiceLocatorConnector.scala]: unused imports

File: app/uk/gov/hmrc/helptosaveapi/connectors/ServiceLocatorConnector.scala [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+class ServiceLocatorConnectorImpl @Inject() (config:      Configuration,
+                                             http:        WSHttp,
+                                             environment: Environment) extends ServiceLocatorConnector with ServicesConfig {
+
+  val mode: Mode = environment.mode
+
+  override val runModeConfiguration: Configuration = config
+
+  private val appName: String = getString("appName")
+  private val appUrl: String = getString("appUrl")
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/helptosaveapi/connectors/ServiceLocatorConnector.scala]: do we need to override the `appUrl` in different environments ?

File: test/resources/public.api.conf.1.0/application.raml [hunk 5]
```diff
@@ -0,0 +1,161 @@
+#%RAML 1.0
```
Comment (sureshhmrc) [inline test/resources/public.api.conf.1.0/application.raml]: why is this only in `test` folder ?  where is actual raml gets stored ?  or not required ?

File: app/uk/gov/hmrc/helptosaveapi/ApplicationRegistration.scala (hunk 1 above)
Comment (Jackie1988) [reply app/uk/gov/hmrc/helptosaveapi/ApplicationRegistration.scala]: removed, thanks

File: app/uk/gov/hmrc/helptosaveapi/ApplicationRegistration.scala (hunk 2 above)
Comment (Jackie1988) [reply app/uk/gov/hmrc/helptosaveapi/ApplicationRegistration.scala]: Thats taken care of in the ServiceLocatorConnector

File: app/uk/gov/hmrc/helptosaveapi/connectors/ServiceLocatorConnector.scala (hunk 3 above)
Comment (Jackie1988) [reply app/uk/gov/hmrc/helptosaveapi/connectors/ServiceLocatorConnector.scala]: removed.

File: app/uk/gov/hmrc/helptosaveapi/connectors/ServiceLocatorConnector.scala (hunk 4 above)
Comment (Jackie1988) [reply app/uk/gov/hmrc/helptosaveapi/connectors/ServiceLocatorConnector.scala]: Yes I would have thought so

File: test/resources/public.api.conf.1.0/application.raml (hunk 5 above)
Comment (Jackie1988) [reply test/resources/public.api.conf.1.0/application.raml]: The raml file is located in resources/public/api/conf/1.0 folder, I couldnt seem to access this from the tests so I copied the raml file to the test/resources/public/api/conf/1.0 as well

File: app/uk/gov/hmrc/helptosaveapi/ApplicationRegistration.scala (hunk 2 above)
Comment (sureshhmrc) [reply app/uk/gov/hmrc/helptosaveapi/ApplicationRegistration.scala]: ServiceLocatorConnector returns false if there is any error during registration, and `this` false case will be caught by `case Success(result)` and logs as service successfully registered which is misleading

Comment (AndyHWChung) [reply app/uk/gov/hmrc/helptosaveapi/ApplicationRegistration.scala]: here we just need something like this:
```scala
  case Success(result) =>
     if(result){
      logger.info("Registration with service locator successful")
    } else {
     logger.warn("Registration with service locator unsuccessful")
   }
...
```

Comment (Jackie1988) [reply app/uk/gov/hmrc/helptosaveapi/ApplicationRegistration.scala]: Now I understand, added thanks

File: test/resources/public.api.conf.1.0/application.raml (hunk 5 above)
Comment (Jackie1988) [reply test/resources/public.api.conf.1.0/application.raml]: Sorted thanks to Andy!!

**Comment by Jackie1988** [comment]
closed in favour of https://github.com/hmrc/help-to-save-api/pull/9

**Pull request closed without merging**